src/discord/discord.go

```diff
@@ -0,0 +1,25 @@
+package discord
+
+import (
+	"net/http"
+	"encoding/json"
+	"fmt"
+	"bytes"
+
+	"github.com/devclub-iitd/DeployBot/src/history"
+	log "github.com/sirupsen/logrus"
+)
+
+func PostActionMessage(a *history.ActionInstance , callbackID string) error {
+	msg := newMessage(a, callbackID)
+	log.Infof("Posting chat message to discord")
+	payload, _ := json.Marshal(msg)
+	resp, err := http.Post(postMessageHookURL, "application/json", bytes.NewBuffer(payload))
```

Comment:
Are you sure it is this simple? Do you not have any authorization headers?
How does discord know, where to post and who is posting?

Reply:
Discord generates a unique webhook url for posting in a channel on a server. It requires no authorization.